docs/using-est-with-certd.md

```diff
@@ -0,0 +1,187 @@
+# Optional if each certificate issuance specifies a URL.
+[cert_issuance.est.urls]
+default = "https://est.example.com/.well-known/est"
+
+# Below are options for authenticating with the EST server. The required options will depend on the EST
+# server's configuration. These global settings apply to all certificates that don't configure auth separately.
+[cert_issuance.est.auth]
+
+# Authentication with TLS client certificate. Provide the cert ID of the client cert and its corresponding
+# private key. Note that the aziotcs user must be authorized to access `identity_pk` in Keys Service.
+identity_cert = "identity-cert-id"
+identity_pk = "identity-cert-pk-id"
+
+# Authentication with a TLS client certificate which will be used once to create the initial certificate.
+# After the first certificate issuance, an identity_cert and identity_pk will be automatically created and
+# used. Provide the cert ID of the bootstrap client cert and its corresponding private key. Note that the
+# aziotcs user must be authorized to access `bootstrap_identity_pk` in Keys Service.
+bootstrap_identity_cert = "bootstrap-identity-cert-id"
+bootstrap_identity_pk = "bootstrap-identity-pk-id"
+
+# Authentication with username and password.
+username = "username"
+password = "password"
+
+# Sample configuration of a single EST-issued certificate.
+# Replace `name` with the desired certificate name.
+[cert_issuance.name]
+
+# Identifies this certificate as being issued over EST.
+method = "est"
+
+# Optional certificate common name. Defaults to the CSR's common name if not provided.
+common_name = "common name"
+
+# Optional number of days between certificate issuance and expiry. Defaults to 30 if not provided.
+expiry_days = 30
+
+# Optional EST URL to issue this certificate. Defaults to the `default` URL in `[cert_issuance.est.urls]`
+# if not provided. The URL must be provided either here or in default, i.e. certd will fail if no URL is
+# provided here and no default exists.
+url = "https://est.example.com/.well-known/est"
+
+# It is also possible to configure auth separately for each certificate. The options are the
+# same as in the global EST configuration and override the global configuration for their corresponding
+# certificate.
+identity_cert = "identity-cert-id"
+identity_pk = "identity-cert-pk-id"
+
+bootstrap_identity_cert = "bootstrap-identity-cert-id"
+bootstrap_identity_pk = "bootstrap-identity-pk-id"
+
+username = "username"
+password = "password"
+```
+
+## Sample Configuration
+
+Below are sample configuration files that can be used as starting points to test EST certificate issuance.
+
+The certd configuration file configures EST certificate issuance.
+
+`/etc/aziot/certd/config.d/test-est-cert.toml`
+
+```toml
+# Configure the trusted root CA certificate in the global EST options. This section is optional
+# if the EST server's TLS certificate is already trusted by the system's CA certificates.
+[cert_issuance.est]
+trusted_certs = ["est-server-ca"]
```

Comment:
Can this also be specified by inlining the file name like this? https://github.com/Azure/iot-identity-service/blob/main/aziotctl/aziotctl-common/test-files/apply/dps-x509-pkcs11-est-bootstrap/config.toml#L16

Reply:
No, it's not possible for this file. It needs to be specified as a cert ID (with the file then loaded under preloaded_certs).

Reply:
so do we need to change that test config.toml?

Reply:
No, that test file is a super config. The file:// URI format for that file works.